### DISCUSSION TITLE
Only able to use teleport via client with --insecure

### DISCUSSION THREAD

**Discussion by Biz0r** [gravitational/teleport#21503] [Q&A]
Hello guys,

I have a issue which I cant fix right now.

I have a reverse proxy (nginx) forwarding :3080 to :443 which works fine.

If I use teleport.example.com in webbrowser, I can connect without any issues.
When I try to use the tsh login --proxy=teleport.example.com --user=exampleuser... it won't work until I use --insecure.
The error is that my cert is not "valid".
But it is and I'm not able to get it work.

I also tried this without reverse proxy :3080 and directly via docker.

Has someone an idea?

Best regards,
Domi

**Comment by partoche** [comment]
Hello, 

tsh will use the computer's certificate stores  (user store and machine store) to validate certificates.

If your browser is firefox, it may have its own certificate store.

It depends if you are on windows or linux.

If you are on windows, the connection should show the same warning with Edge or Internet Explorer.
If you do, open your webbrowser that does not show the error, connect to your instance, show the certificate and export the authority to a file, open certmgr.msc and import the authority in the "Authorities".


If you are on linux, you can open your web browser, connect to your teleport instance, then view the certificate and its authority, export the authority to a file, and import it into the ca-certificates (https://ubuntu.com/server/docs/security-trust-store)

**Comment by webvictim** [comment]
@Biz0r Do you have `proxy_listener_mode: multiplex` in your `/etc/teleport.yaml` config file?

Is your nginx reverse proxy also injecting the SSL certificate that you're using to secure Teleport, or do you have this set using `https_keypairs` in the Teleport config file?

**Reply by Hamadn** [reply]
i am having this same issue, the proxy is injecting SSL and the teleport instance has a secure connection. but when i add an application to teleport and click the link, it shows a connection not secure page and is also not letting me access the website.
when i go to certificates, it says it is from TRAEFIK and it is an unknown authority.

**Reply by webvictim** [reply]
This sounds like your Traefik instance is not configured to provide a valid certificate for the Teleport application's subdomain.

For example, it has a certificate for `teleport.example.com` but not `app.teleport.example.com`
